apps/opencs/view/render/cellarrow.cpp

```diff
@@ -0,0 +1,155 @@
+{
+    // position
+    const int cellSize = 8192;
+    const int offset = cellSize / 2 + 800;
+
+    int x = mCoordinates.getX()*cellSize + cellSize/2;
+    int y = mCoordinates.getY()*cellSize + cellSize/2;
+
+    float xr = 0;
+    float yr = 0;
+    float zr = 0;
+
+    float angle = osg::DegreesToRadians (90.0f);
+
+    switch (mDirection)
+    {
+        case Direction_North: y += offset; xr = -angle; zr = angle; break;
+        case Direction_West: x -= offset; yr = -angle; break;
+        case Direction_South: y -= offset; xr = angle; zr = angle; break;
+        case Direction_East: x += offset; yr = angle; break;
+    };
+
+    mBaseNode->setPosition (osg::Vec3f (x, y, 0));
+
+    // orientation
+    osg::Quat xr2 (xr, osg::Vec3f (1,0,0));
+    osg::Quat yr2 (yr, osg::Vec3f (0,1,0));
+    osg::Quat zr2 (zr, osg::Vec3f (0,0,1));
+    mBaseNode->setAttitude (zr2*yr2*xr2);
+}
+
+void CSVRender::CellArrow::buildShape()
+{
+    osg::ref_ptr<osg::Geometry> geometry (new osg::Geometry);
+
+    const int arrowWidth = 4000;
+    const int arrowLength = 1500;
+    const int arrowHeight = 500;
+
+    osg::Vec3Array *vertices = new osg::Vec3Array;
+    for (int i2=0; i2<2; ++i2)
+        for (int i=0; i<2; ++i)
+        {
+            float height = i ? -arrowHeight/2 : arrowHeight/2;
+            vertices->push_back (osg::Vec3f (height, -arrowWidth/2, 0));
+            vertices->push_back (osg::Vec3f (height, arrowWidth/2, 0));
+            vertices->push_back (osg::Vec3f (height, 0, arrowLength));
+        }
+
+    geometry->setVertexArray (vertices);
+
+    osg::DrawElementsUShort *primitives = new osg::DrawElementsUShort (osg::PrimitiveSet::TRIANGLES, 0);
+
+    // top
+    primitives->push_back (0);
+    primitives->push_back (1);
+    primitives->push_back (2);
+
+    // bottom
+    primitives->push_back (5);
+    primitives->push_back (4);
+    primitives->push_back (3);
+
+    // back
+    primitives->push_back (3+6);
+    primitives->push_back (4+6);
+    primitives->push_back (1+6);
+
+    primitives->push_back (3+6);
+    primitives->push_back (1+6);
+    primitives->push_back (0+6);
+
+    // sides
+    primitives->push_back (0+6);
+    primitives->push_back (2+6);
+    primitives->push_back (5+6);
+
+    primitives->push_back (0+6);
+    primitives->push_back (5+6);
+    primitives->push_back (3+6);
+
+    primitives->push_back (4+6);
+    primitives->push_back (5+6);
+    primitives->push_back (2+6);
+
+    primitives->push_back (4+6);
+    primitives->push_back (2+6);
+    primitives->push_back (1+6);
+
+    geometry->addPrimitiveSet (primitives);
+
+    osg::Vec4Array *colours = new osg::Vec4Array;
+
+    for (int i=0; i<6; ++i)
+        colours->push_back (osg::Vec4f (1.0f, 0.0f, 0.0f, 1.0f));
+    for (int i=0; i<6; ++i)
+        colours->push_back (osg::Vec4f (0.8f, (i==2 || i==5) ? 0.6f : 0.4f, 0.0f, 1.0f));
+
+    geometry->setColorArray (colours);
+    geometry->setColorBinding (osg::Geometry::BIND_PER_VERTEX);
```

Comment:
That doesn't build for me. 

geometry->setColorArray (colours, osg::Array::BIND_PER_VERTEX);

does though. Fixing ...